Support customization / props of submit button using ui:submitButtonOptions (#2640)
rjsf-team/react-jsonschema-form: docs/api-reference/uiSchema.md

{
  "ui:title": "Title",
  "ui:description": "Description"
  "ui:submitButtonOptions": {
    "props": {
      "disabled": false,
      "className": "btn btn-info",
    },
      "removed": false,
      "submitText": "Submit"
    },
}
```

```json
{
  "ui:options": {
    "title": "Title",
    "description": "Description",
    "submitButtonOptions": {

>       props: {

Reviewer: ```suggestion
      "props": {
```

  "ui:title": "Your password"
};
```

## submitButtonProps

Sometimes it's convenient to change the behavior of the submit button for the form. This is the purpose of the `ui:submitButtonProps` uiSchema directive:

You can pass any other prop to the submit button if you want, by default, this library will set the following options mentioned below for all submit buttons:

### `disabled` option

You can use this option to disable the submit button

### `className` option

You can use this option to specify a class name for the submit button

### `required` option

> You can use this option to remove the submit button completely from the form. Nice option, if the form is just for viewing purposes

Reviewer: Hi @jacqueswho , I'm not comfortable calling this `required` because `required` has a different meaning otherwise in json schema (usually it just means that the field will be shown, but not required), but here, it basically means that the button will just be removed. Why not call it `hidden` instead?

Also, do you want to have a quick video call so we can review this PR faster and in real time? I hope this will speed up reviews :) https://calendly.com/aramaswamis/meeting?month=2022-01

  "ui:title": "Your password"
};
```

## submitButtonProps

Sometimes it's convenient to change the behavior of the submit button for the form. This is the purpose of the `ui:submitButtonProps` uiSchema directive:

By default, this library will set the following options mentioned below for all submit buttons:
### `hidden` option

You can use this option to hide the submit button

### `disabled` option

You can use this option to disable the submit button

### `className` option

You can use this option to specify a class name for the submit button

> ### `required` option

Reviewer: I think we don't need the `required` option because the `hidden` option already accomplishes this purpose?

Author: hidden is if you still want to have it in the dom. required removes it completely. Also hidden will only probably work on some browsers.

Author: I have removed the hidden option. mentioned in the docs they can pass any other prop they want, so hidden if they want to

  "ui:title": "Your password"
};
```

## submitButtonOptions

Sometimes it's convenient to change the behavior of the submit button for the form. This is the purpose of the `ui:submitButtonOptions` uiSchema directive:

You can pass any other prop to the submit button if you want, by default, this library will set the following options / props mentioned below for all submit buttons:

### `disabled` prop

You can use this option to disable the submit button

> ### `className` prop

Reviewer: We should remove `disabled`, `className` sections because these are all now part of the `props` property. We should also add a section that says

```suggestion
### `props` prop
```

and describes what it does.

> You can use this option to specify a class name for the submit button

Reviewer: ```suggestion
You can use this option to specify a class name for the submit button.
```

### `removed` option

> You can use this option to remove the submit button completely from the form. Nice option, if the form is just for viewing purposes

Reviewer: ```suggestion
You can set this property to `true` to remove the submit button completely from the form. Nice option, if the form is just for viewing purposes.
```

  "ui:title": "Your password"
};
```

## submitButtonOptions

Sometimes it's convenient to change the behavior of the submit button for the form. This is the purpose of the `ui:submitButtonOptions` uiSchema directive:

You can pass any other prop to the submit button if you want, by default, this library will set the following options / props mentioned below for all submit buttons:

> ### `removed` option

Reviewer: As per discussion in 2/25 call, let's rename this to:

```suggestion
### `norender` option
```

this is more intuitive to the user and reflects how it's actually implemented (button is not rendered)

You can set this property to `true` to remove the submit button completely from the form. Nice option, if the form is just for viewing purposes.

### `submitText` option

> You can use this option to change the text of the submit button. Set to "submit" by default.

Reviewer: ```suggestion
You can use this option to change the text of the submit button. Set to "Submit" by default.
```

### `props` section

You can pass any other prop to the submit button if you want, via this section.

> 

Reviewer: ```suggestion
```

####  `disabled` prop

You can use this option to disable the submit button.

#### `className` prop

You can use this option to specify a class name for the submit button.

```js
const uiSchema = {

>  "submitButtonOptions": {

Reviewer: ```suggestion
 "ui:submitButtonOptions": {
```
